internal/pipeline/command_delete.go

```diff
@@ -48,18 +48,29 @@ func (c *command) delete(cmd *cobra.Command, args []string) error {
 
 	pipeline, err := c.V2Client.GetSdPipeline(environmentId, cluster.ID, args[0])
 	if err != nil {
-		return err
+		return resource.ResourcesNotFoundError(cmd, resource.Pipeline, args[0])
 	}
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmMsg, resource.Pipeline, pipeline.GetId(), pipeline.Spec.GetDisplayName())
-	if _, err := form.ConfirmDeletion(cmd, promptMsg, pipeline.Spec.GetDisplayName()); err != nil {
-		return err
+	existenceFunc := func(id string) bool {
+		_, err := c.V2Client.GetSdPipeline(environmentId, cluster.ID, id)
+		return err == nil
 	}
 
-	if err := c.V2Client.DeleteSdPipeline(environmentId, cluster.ID, args[0]); err != nil {
+	if err := deletion.ValidateAndConfirmDeletion(cmd, args, existenceFunc, resource.Pipeline, pipeline.Spec.GetDisplayName()); err != nil {
 		return err
 	}
 
-	output.Printf(errors.RequestedDeleteResourceMsg, resource.Pipeline, args[0])
-	return nil
+	deleteFunc := func(id string) error {
+		return c.V2Client.DeleteSdPipeline(environmentId, cluster.ID, id)
+	}
+
+	deletedIds, err := deletion.DeleteWithoutMessage(args, deleteFunc)
+	deleteMsg := "Requested to delete %s %s.\n"
+	if len(deletedIds) == 1 {
+		output.Printf(deleteMsg, resource.Pipeline, fmt.Sprintf("\"%s\"", deletedIds[0]))
```

Comment:
Hmm, I did this because the plural version calls `ArrayToCommaDelimitedString`, which puts quotation marks around everything, and I wanted it to be consistent.

Perhaps I should remove the quotations from that function in a followup PR, because it's not actually what we would expect to happen just based on the name...

Reply:
Lol oops, I completely misunderstood.